### ISSUE TITLE
[Bug] App crashes on mac when clicking on profile tab

### ISSUE THREAD

**Issue by setch-l** [planetary-social/nos#778]
Clicking on the Profile tab crashes the app in build 145 in production on a mac.
Steps to reproduce:
Open nos.social on a mac
Log out of account
Log in with another npubid
Create a test post
Click on profile
Issue: App crashes.

Log files and a corresponding Sentry report can be found in this thread in slack: https://planetary-app.slack.com/archives/CM4EPK324/p1704381641479009

**Comment by mplorentz** [comment]
I did some investigation on this one, but haven't found a solution yet. @setch-l I'm not able to reproduce this one with the above steps, are you? If so a screen recording would be great. 

The associated [Sentry report](https://sentry.nos.social/organizations/verse/issues/774/?alert_rule_id=3&alert_type=issue&project=2&referrer=slack) actually crashed in the code that is performed when the Follow button is pressed. The stack trace on that crash is really weird, but it seems to be happening in a call to NSManagedObjectContext.save(). I found [another crash report](https://sentry.nos.social/organizations/verse/issues/742/events/951a48642f4e4096888ee9b36f4d61d2/?query=is%3Aunresolved+error.unhandled%3Atrue&referrer=previous-event&statsPeriod=14d&stream_index=0) that is also happening on save that has a more useful stack trace. It seems to be a problem handling deletes in PagedNoteDataSource. I haven't been able to reproduce it yet but for now I am assuming these are the same crash. I'll continue looking for steps to reproduce and a fix next week.
